packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -58,14 +72,46 @@ export class EditorSettingsSidebarComponent {
 	 * @param {EditorSidebarSection} sectionName Name of section.
 	 * @returns {Promise<void>} No return value.
 	 */
-	async expandSectionIfCollapsed( sectionName: EditorSidebarSection ): Promise< void > {
+	async expandSection( sectionName: EditorSidebarSection ): Promise< void > {
 		// Avoid the wpcalypso/staging banner
 		await this.scrollToBottomOfSidebar();
 		if ( ! ( await this.frame.isVisible( selectors.expandedSection( sectionName ) ) ) ) {
 			await this.frame.click( selectors.sectionToggle( sectionName ) );
 		}
 	}
 
+	/* Status & Visibility */
+
+	/**
+	 * Sets the post visibility to the provided visibility setting.
+	 *
+	 * @param {PostVisibilityOptions} visibility Desired post visibility setting.
+	 */
+	async setVisibility( visibility: PostVisibilityOptions ): Promise< void > {
+		await this.expandSection( 'Status & Visibility' );
+		await this.frame.click( selectors.visibilityToggle );
+		// Important to wait for the popover element to finish its animation, as the radio buttons
+		// will not be actionable until then.
+		const popoverHandle = await this.frame.waitForSelector( selectors.visibilityPopover );
+		await popoverHandle.waitForElementState( 'stable' );
+
+		// Private posts display a dialog that when accepted will publish the post.
+		// For non-Private posts, this handler has no effect.
```

Comment:
Should we just make this explicit and check if we're making it private, and add the handler only then?

Reply:
We can, though as noted in the comments, the `page.on` handler is essentially a free call. I felt that it was preferable to present a 'unified' behavior expectation instead of a conditional, which may change later down the road.

Reply:
Ah sure, that's a great counter point! Works for me 👍